packages/notifi-react/lib/context/NotifiTargetContext.tsx

```diff
@@ -669,14 +673,7 @@ export const NotifiTargetContextProvider: FC<PropsWithChildren> = ({
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore
       const signature = await walletWithSignParams.signMessage(message);
-      let hexString = '0x';
-
-      for (let i = 0; i < Object.keys(signature).length; i++) {
-        // Convert each value to a two-digit hexadecimal string and append it to the result
-        hexString += signature[i].toString(16).padStart(2, '0');
-      }
-
-      return hexString;
+      return reformatSignatureForWalletTarget(signature);
```

Comment:
I think we need to get rid of the type issue for scalability and maintainability as it is SDK.
I think this type error is from mis handling chain specific differences. We might want to handle those case, just an conceptual example:
```ts
switch (walletWithSignParams.walletBlockchain) {
        case 'AVALANCHE':
        case 'ETHEREUM':
        case ...etc:
        // handle get signature
        break;
        default:
        // handle error
      }
```